File: packages/frontend/src/api/network.api.ts
import api from './client';
import type { ApiResponse } from '@/types';

export interface StoryUser {
  id: string;
  name: string;
  businessName?: string;
  newCount: number;
}

export interface ConnectionUser {
  id: string;
  name: string;
  businessName?: string;
  role: string;
  phone?: string;
}

export interface TraderInsightsResponse {
  trader: ConnectionUser;
  stats: {
    ordersCount: number;
    uniqueBuyersFromOrders: number;
    buyersFollowingTrader: number;
    curatedShareRecipients: number;
    activeProductsWithTrader: number;
  };
}

export const networkApi = {
  getStories: () =>
    api.get<ApiResponse<StoryUser[]>>('/network/stories').then((r) => r.data.data),

  getConnections: () =>
    api.get<ApiResponse<ConnectionUser[]>>('/network/connections').then((r) => r.data.data),

  getSuggestions: () =>
    api.get<ApiResponse<ConnectionUser[]>>('/network/suggestions').then((r) => r.data.data),

  follow: (userId: string) =>
    api.post<ApiResponse<unknown>>(`/network/follow/${userId}`).then((r) => r.data),

  unfollow: (userId: string) =>
    api.delete<ApiResponse<unknown>>(`/network/unfollow/${userId}`).then((r) => r.data),

  search: (q: string) =>
    api.get<ApiResponse<ConnectionUser[]>>('/network/search', { params: { q } }).then((r) => r.data.data),

  getInviteCode: () =>
    api.get<ApiResponse<{ code: string }>>('/network/invite-code').then((r) => r.data.data),

  connectViaInvite: (code: string) =>
    api.post<ApiResponse<{ id: string; name: string; businessName?: string }>>('/network/connect-invite', { code }).then((r) => r.data.data),

  /** Vendor: link an existing trader on GarmentHub (search). Use share invite for people not on the app. */
  connectTrader: (traderId: string) =>
    api.post<ApiResponse<unknown>>(`/network/connect-trader/${traderId}`).then((r) => r.data),

  getTraderInsights: (traderId: string) =>
    api.get<ApiResponse<TraderInsightsResponse>>(`/network/traders/${traderId}/insights`).then((r) => r.data.data),
};

File: packages/mobile/lib/core/api/network_api.dart
import 'package:dio/dio.dart';

import '../network/api_response.dart';

class NetworkApi {
  NetworkApi(this._dio);
  final Dio _dio;

  Future<List<dynamic>> getStories() async {
    final res = await _dio.get<Map<String, dynamic>>('/network/stories');
    return ApiResponse.unwrapData(res.data!, (d) => d as List<dynamic>);
  }

  Future<List<dynamic>> getConnections() async {
    final res = await _dio.get<Map<String, dynamic>>('/network/connections');
    return ApiResponse.unwrapData(res.data!, (d) => d as List<dynamic>);
  }

  Future<List<dynamic>> getSuggestions() async {
    final res = await _dio.get<Map<String, dynamic>>('/network/suggestions');
    return ApiResponse.unwrapData(res.data!, (d) => d as List<dynamic>);
  }

  Future<void> follow(String userId) async {
    await _dio.post<Map<String, dynamic>>('/network/follow/$userId');
  }

  Future<void> unfollow(String userId) async {
    await _dio.delete<Map<String, dynamic>>('/network/unfollow/$userId');
  }

  Future<List<dynamic>> search(String q) async {
    final res = await _dio.get<Map<String, dynamic>>('/network/search', queryParameters: {'q': q});
    return ApiResponse.unwrapData(res.data!, (d) => d as List<dynamic>);
  }

  Future<Map<String, dynamic>> getInviteCode() async {
    final res = await _dio.get<Map<String, dynamic>>('/network/invite-code');
    return ApiResponse.unwrapData(res.data!, (d) => d as Map<String, dynamic>);
  }

  Future<Map<String, dynamic>> connectViaInvite(String code) async {
    final res = await _dio.post<Map<String, dynamic>>('/network/connect-invite', data: {'code': code});
    return ApiResponse.unwrapData(res.data!, (d) => d as Map<String, dynamic>);
  }

  Future<void> connectTrader(String traderId) async {
    await _dio.post<Map<String, dynamic>>('/network/connect-trader/$traderId');
  }

  Future<Map<String, dynamic>> getTraderInsights(String traderId) async {
    final res = await _dio.get<Map<String, dynamic>>('/network/traders/$traderId/insights');
    return ApiResponse.unwrapData(res.data!, (d) => d as Map<String, dynamic>);
  }
}
